pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -650,6 +698,35 @@ func (r *ReconcileClusterPool) setMissingDependenciesCondition(pool *hivev1.Clus
 	return nil
 }
 
+func (r *ReconcileClusterPool) setAvailableCapacityCondition(pool *hivev1.ClusterPool, available bool, logger log.FieldLogger) error {
+	status := corev1.ConditionFalse
+	reason := "Available"
+	message := "There is capacity to add more clusters to the pool."
+	updateConditionCheck := controllerutils.UpdateConditionNever
+	if !available {
+		status = corev1.ConditionTrue
+		reason = "NoMoreCapacity"
+		message = fmt.Sprintf("There is no more capacity available for the pool. Additional capacity will be required to fulfil the requirements.")
```

Comment:
s/fulfil/fulfill but would suggest maybe a little more concise: "Pool is at maximum capacity of %d waiting and claimed clusters."